chat-android/src/main/java/com/ably/chat/FlowUtils.kt

```diff
@@ -0,0 +1,22 @@
+package com.ably.chat
+
+import kotlinx.coroutines.channels.Channel
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.flow
+
+/**
+ * Converts a callback-based API to a Flow with an unlimited buffer to handle backpressure
+ */
+internal inline fun <T> transformCallbackAsFlow(crossinline subscribe: ((T) -> Unit) -> Subscription): Flow<T> = flow {
+    val channel = Channel<T>(Channel.UNLIMITED)
```

Comment:
We need to close this channel once unsubscribed. As per guidelines
Preventing Resource Leaks: If a channel is not closed, it might remain open indefinitely, potentially leading to resource leaks or unnecessary resource usage.

Reply:
https://stackoverflow.com/questions/59412793/kotlin-coroutines-channel-vs-flow
https://www.baeldung.com/kotlin/flows-vs-channels

Reply:
shouldn't be any leak hear, channel has pure in-memory implementation, will be garbage collected without any problem. But I replaced implementation with `callbackFlow` builder to hide channel